### PR TITLE
Release/v0.9.3.3

### PR DESCRIPTION
**Pull request by davidpofo**
v0.9.3.3 (April 13, 2021)
---------------------

**Feature changes**

* Added support for Remote Interpreter on IDEs for the local Docker deployment.

**UI changes**

* Add "Help" link to global navbar.
* Remove caret glyphicons from global navbar.
* Make conditional admin "Settings" link in global navbar a dropdown menu to include link to Django database admin.
* Simplify task-finished page layout. Move navigation buttons to top.
* Replace "...and we're done" language with "Module Summary".
* Replace questions progress sidebar's project links with more obvious project buttons.

**Bug fixes**

* User now has the ability to edit uploaded files via the admin panel.
* File names now updated properly for all Asset models in the event of an update.
* Added a short README.md to each `modules/systems` folder (account, organization) to avoid seeing the README error when loading modules.

**Developer changes**
* (fields.W903) NullBooleanField is deprecated. Support for it (except in historical migrations) will be removed in Django 4.0. Using BooleanField instead for `siteapp.Project.is_organization_project` and `guidedmodules.AppVersion.system_app`.
* Added version data for the project and the project's compliance app to the exported project json.

**Install fixes**

* Create portfolios for admins when passing in ADMIN setting for automated admin creation during install first-run.
* Create default org 'main' if none exists earlier in the first-run process.
* Fix adding admin user to Help Squad and Reviewers list.
* Install default AppSources and compliance apps only if no AppSources installed.

### CODE REVIEW

File: siteapp/forms.py [hunk 1]
```diff
@@ -5,15 +5,23 @@
...
 
-class ProjectForm(ModelForm):
+class EditProjectForm(ModelForm):
+    class Meta:
+        model = Project
+        fields = ['root_task']
+        project_title = forms.CharField(label='Project Title', max_length=200)
+        project_version = forms.CharField(label='Project Version', max_length=200)
+
+
```
Comment (peterkaminski) [inline siteapp/forms.py:15]: maybe delete extra blank linke

File: siteapp/model_mixins/tags.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
+from django.db import models
```
Comment (peterkaminski) [inline siteapp/model_mixins/tags.py:1]: several places with extra blank lines in this file - maybe delete?

File: siteapp/serializers.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+from rest_framework import permissions
+from rest_framework import serializers
+from rest_framework import viewsets
+from siteapp.models import Project, User
+
+
```
Comment (peterkaminski) [inline siteapp/serializers.py:6]: maybe delete extra blank line?

File: siteapp/urls.py [hunk 4]
```diff
@@ -97,6 +117,11 @@
     url(r'^health/request-headers$', views_health.request_headers),
     url(r'^health/request$', views_health.request),
     url(r'^health/debug$', views.debug, name="debug"),
+
+
```
Comment (peterkaminski) [inline siteapp/urls.py:121]: 1. maybe delete extra blank line?

2. add a "# tags" comment line to group the routes

File: siteapp/views.py [hunk 5]
```diff
@@ -154,13 +164,25 @@ def subvars(s):
...
+    serializer_class = UserSerializer
+
+class ProjectViewSet(viewsets.ModelViewSet):
+    url = serializers.HyperlinkedIdentityField(view_name="siteapp:task-detail")
+
+    queryset = Project.objects.all()
+    serializer_class = ProjectSerializer
+
+
+
```
Comment (peterkaminski) [inline siteapp/views.py:178]: maybe delete blank lines?

File: templates/authoring-tool-modal.html [hunk 6]
```diff
@@ -6,11 +6,12 @@
 
 
 /* remove modal shadow */
-div.modal-content{
-    -webkit-box-shadow: none;
+div.modal-content-authoring {
+   -webkit-box-shadow: none;
```
Comment (peterkaminski) [inline templates/authoring-tool-modal.html:10]: maybe fix indentation?

File: templates/authoring-tool-modal.html [hunk 7]
```diff
@@ -6,11 +6,12 @@
...
     -moz-box-shadow: none;
     -o-box-shadow: none;
     box-shadow: none;
+
```
Comment (peterkaminski) [inline templates/authoring-tool-modal.html:14]: is this blank line needed or useful?

**Review by peterkaminski — COMMENTED** [review]
mostly LGTM; i had one question about the expiry on the cryptography exception for Snyk, and a bunch of nit-picky comments about multiple blank lines and an indentation thing.

the extra blank lines aren't fatal, but they make the code look a little messier to me.  YMMV.

File: .snyk [hunk 8]
```diff
@@ -0,0 +1,9 @@
+# Snyk (https://snyk.io) policy file, patches or ignores known vulnerabilities.
+version: v1.19.0
+ignore:
+  SNYK-PYTHON-CRYPTOGRAPHY-1022152:
+    - '*':
+        reason: >-
+          Patched to version 3.4.7; long term fix expected later.
+        expires: 2022-04-10T06:00:00.000Z
```
[the remark this answers is not in the document]
Comment (davidpofo) [reply .snyk:8]: yeah for now 1 year should be okay. When there is a new release this ignore would be removed. This is avoid having to encounter the erroneous error until a patches fixes the vulnerability.

**Review by peterkaminski — APPROVED** [review]
thanks! looks good

**Pull request merged**